documentation/modules/oauth/con-oauth-authentication-flow.adoc

```diff
@@ -1,12 +1,91 @@
+= OAuth 2.0 authentication mechanisms
 
-The Kafka _SASL OAUTHBEARER_ mechanism is used to establish authenticated sessions with a Kafka broker.
+Strimzi supports the OAUTHBEARER and PLAIN mechanisms for OAuth 2.0 authentication. 
+Both mechanisms allow Kafka clients to establish authenticated sessions with Kafka brokers. 
+The authentication flow between clients, the authorization server, and Kafka brokers is different for each mechanism.
 
-A Kafka client initiates a session with the Kafka broker using the _SASL OAUTHBEARER_ mechanism for credentials exchange, where credentials take the form of an access token.
+Configure clients to use OAUTHBEARER whenever possible. 
+This mechanism provides a higher level of security because client credentials are _never_ shared with Kafka brokers. 
+Consider using PLAIN only with Kafka clients that do not support OAUTHBEARER. 
 
-Kafka brokers and clients need to be configured to use OAuth 2.0.
+If necessary, OAUTHBEARER and PLAIN can be enabled together, on the same `oauth` listener.
+
+.OAUTHBEARER overview
+
+OAUTHBEARER is the default authentication mechanism included as part of Kafka.
+
+Many Kafka client tools use libraries that provide basic support for OAUTHBEARER at the protocol level. 
+To ease application development, Strimzi provides an _OAuth callback handler_ for the upstream Kafka Client Java libraries (but not for other libraries). 
+An application client can use the callback handler to provide the access token. 
+Clients written in other languages, such as Go, must use custom code to connect to the authorization server and obtain the access token.
+
+With OAUTHBEARER, the client initiates a session with the Kafka broker for credentials exchange, where credentials take the form of a bearer token provided by the callback handler. 
+Using the callbacks, you can configure token provision in one of three ways:
+
+* Client ID and Secret (by using the OAuth 2.0 client credentials mechanism)
+
+* A long-lived access token, obtained manually at configuration time
+
+* A long-lived refresh token, obtained manually at configuration time
+
+OAUTHBEARER is automatically enabled in the `oauth` listener configuration for the Kafka broker. 
+You can set the `enableOauthBearer` property to `true`, though this is not required.
+
+[source,yaml,subs="attributes+"]
+----
+  # ...
+  authentication:
+    type: oauth
+    # ...
+    enableOauthBearer: true
+----
+
+[NOTE]
+====
+OAUTHBEARER authentication can only be used by Kafka clients that support the OAUTHBEARER mechanism at the protocol level.
+====
+
+.PLAIN overview
+
+PLAIN is a simple authentication mechanism used by all Kafka client tools (including developer tools such as kafkacat). 
+To enable PLAIN to be used together with OAuth 2.0 authentication, Strimzi includes server-side callbacks and calls this _OAuth 2.0 over PLAIN_. 
+
+With PLAIN, client credentials are handled centrally, behind a compliant authorization server.
```

Comment:
```suggestion
With Strimzi implementation of PLAIN, the client credentials are not stored in Zookeeper but are handled centrally, behind a compliant authorization server, just like when OAUTHBEARER is used.
```

Reply:
I applied the suggestion but reworded slightly.